lib/downloader.rb

```diff
@@ -21,6 +21,28 @@ def downloader (url, filename = File.basename(url), retry_count = 0, verbose = f
   trap('WINCH') { setTermSize }
   uri = URI(url)
 
+  case uri.scheme
+  when /http?/
```

Comment:
What do we do about `ftp` and other protocols?

Reply:
Fallback to `curl`:

https://github.com/skycocker/chromebrew/pull/6520/files#diff-46c0af844c07a052d4a9dd139bf6e2d494a7032fed363960eaedaf4bc82c9e8dR33-R44

Reply:
It would be more elegant to use net/http for ftp.  Possible?

Reply:
Is https://ruby-doc.org/stdlib-2.7.0/libdoc/open-uri/rdoc/OpenURI.html another option? Since it wraps HTTP and FTP?

Reply:
It is a much better solution :)